add support for version-template (#161)
toolmantim/release-drafter: README.md

| `template`            | Required | The template for the body of the draft release. Use [template variables](#template-variables) to insert values.                                                                                                   |
| `name-template`       | Optional | The template for the name of the draft release. For example: `"v$NEXT_PATCH_VERSION"`.                                                                                                                            |
| `tag-template`        | Optional | The template for the tag of the draft release. For example: `"v$NEXT_PATCH_VERSION"`.                                                                                                                             |

> | `version-template`    | Optional | The template for the version number of the draft release. For example: `"$MAJOR.$MINOR"` will have `"v$NEXT_PATCH_VERSION"` only return `v1.0` used for non-semver version. Default: `"$MAJOR.$MINOR.$PATCH"`     |

Reviewer: ```suggestion
| `version-template`    | Optional | The template for the version number of the draft release. For example: `"$MAJOR.$MINOR"` will have `"v$NEXT_PATCH_VERSION"` only return `v1.0`. Default: `"$MAJOR.$MINOR.$PATCH"`     |
```

| `$NEXT_MINOR_VERSION` | The next minor version number. For example, if the last tag or release was `v1.2.3`, the value would be `v1.3.0`.                                       |
| `$NEXT_MAJOR_VERSION` | The next major version number. For example, if the last tag or release was `v1.2.3`, the value would be `v2.0.0`.                                       |

> ## Version Variables

Reviewer: ```suggestion
## Version Template Variables
```

> You can use any of the following variables in your `version-template` to format your `Next Version` variables:

Reviewer: ```suggestion
You can use any of the following variables in `version-template` to format the `$NEXT_{PATCH,MINOR,MAJOR}_VERSION` variables:
```

<img src="design/screenshot-2.png" alt="Screenshot of generated draft release with categories" width="586" />

> ## Non Semver Versions

Reviewer 2: @TimonVS given we already have first-class concepts of major, minor and patch… reading this in the readme now feels very abstract. Unless we have some other use case examples?

Reviewer: @toolmantim I agree that it might feel a little abstract. I think it's quite a niche feature so maybe we can push it more to the background as a power user feature? The benefit of this more abstract approach is that we can add support for [CalVer](http://calver.org/) easily. I don't know about any other widely used versioning systems though.

Author: Into the background should I move it down inside the README?

Author: Yes would be super easy to add support for CalVer :sweat_smile:

> ## Non Semver Versions

Reviewer: ```suggestion
## Non-SemVer Versions
```

> With `version-template` option you can change the next version format to use your desired version number. For example append the following to your `.github/release-drafer.yml` file:

Reviewer: ```suggestion
With `version-template` option you can change the template of the generated `$NEXT_{PATCH,MINOR,MAJOR}_VERSION` variables. For example append the following to your `.github/release-drafer.yml` file:
```

```yml
version-template: $MAJOR.$MINOR
tag-template: v$NEXT_MINOR_VERSION
```

> If current version is 1.0 the `v$NEXT_MINOR_VERSION` will turn into `v1.1`

Reviewer: ```suggestion
If the current version is 1.0, `v$NEXT_MINOR_VERSION` will turn into `v1.1`.
```

Adding such labels to your PRs can be automated by using [PR Labeler](https://github.com/TimonVS/pr-labeler-action) or [Probot Auto Labeler](https://github.com/probot/autolabeler).

> ## Non-SemVer Versions

Reviewer 2: ## Projects that don't use Semantic Versioning

If your project doesn't follow [Semantic Versioning](https://semver.org) you can still use Release Drafter, but you may want to set the `version-template` option to customize how the `$NEXT_{PATCH,MINOR,MAJOR}_VERSION` environment variables are generated.

For example, if your project doesn't use patch version numbers, you can set `version-template` to `$MAJOR.$MINOR`. If the current release is version 1.0, then `$NEXT_MINOR_VERSION` will be `1.1`.
